### PR TITLE
🎨 Add template option for logo in dark mode

### PR DESCRIPTION
**Pull request by eurunuela**
This PR stems from the conversation in [mystjs #155](https://github.com/executablebooks/mystjs/issues/155).

### CODE REVIEW

File: packages/site/src/components/Navigation/TopNav.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ function ExternalOrInternalLink({
 function NavItem({ item }: { item: SiteNavItem }) {
   if (!('children' in item)) {
     return (
-      <div className="relative grow-0 inline-block mx-2">
+      <div className="relative inline-block mx-2 grow-0">
```
Comment (rowanc1) [inline packages/site/src/components/Navigation/TopNav.tsx:60]: What vscode extension do you use to order tailwind classes? (I am assuming?)

I was looking into/thinking about this today!

File: packages/site/src/components/Navigation/TopNav.tsx [hunk 2]
```diff
@@ -182,19 +182,35 @@ function ActionMenu({ actions }: { actions?: SiteManifest['actions'] }) {
...
   const nothingSet = !logo && !logoText;
   return (
     <Link
-      className="flex items-center dark:text-white w-fit ml-3 md:ml-5 xl:ml-7"
+      className="flex items-center ml-3 dark:text-white w-fit md:ml-5 xl:ml-7"
       to={withBaseurl('/', baseurl)}
       prefetch="intent"
     >
       {logo && (
-        <div className="dark:bg-white dark:rounded p-1 mr-3">
```
Comment (rowanc1) [inline packages/site/src/components/Navigation/TopNav.tsx:196]: I think we might want another branch/if-statement here for the existing behaviour when the dark logo is not defined. I think that should still have the white background?

**Review by rowanc1 — COMMENTED** [review]
This is awesome, glad that you found everything! 🚀 

I will test it out in the next few days, but looks solid, I think we just want a bit more when the dark logo is not provided.

File: packages/site/src/components/Navigation/TopNav.tsx (hunk 1 above)
Comment (eurunuela) [reply packages/site/src/components/Navigation/TopNav.tsx:60]: I use [Headwind](https://marketplace.visualstudio.com/items?itemName=heybourn.headwind).

File: packages/site/src/components/Navigation/TopNav.tsx (hunk 2 above)
Comment (eurunuela) [reply packages/site/src/components/Navigation/TopNav.tsx:196]: Yeah, I was thinking about that too.

Probably something like `logoDark ?? <div className="..."></div>` will work.

Comment (rowanc1) [reply packages/site/src/components/Navigation/TopNav.tsx:196]: The `dark:hidden` class name also needs to be added depending as well.

We have been using something like:

```jsx
<img className={classNames('shared-classes', { 'darkhidden': !!logoDark })} />
```

Comment (eurunuela) [reply packages/site/src/components/Navigation/TopNav.tsx:196]: Oh, that looks much cleaner. Thanks Rowan!

Comment (eurunuela) [reply packages/site/src/components/Navigation/TopNav.tsx:196]: What do you think of https://github.com/executablebooks/myst-theme/pull/130/commits/2e3b14fb79e55448fc9b50b2c518e8047cee912c

**Comment by eurunuela** [comment]
I'm testing these changes but I'm getting this error:

`'options' extra key ignored: logo_dark (at myst.yml)`

Any idea?

File: packages/site/src/components/Navigation/TopNav.tsx (hunk 1 above)
Comment (rowanc1) [reply packages/site/src/components/Navigation/TopNav.tsx:60]: Doing this for the whole project in #135. I will merge after this PR goes in so there are no conflicts.

**Comment by rowanc1** [comment]
Works great -- I updated the logic a bit after testing it out locally.

There is one bit of logic that I am not stoked about, the file URLs need to be updated dynamically and this should happen based on the template, rather than hard-coded. For now I think this is ok, and we can come back and fix this up later when we are copying more files in the template (e.g. custom styles or something).

dark-logo

I also added it to the article theme, and put in a changeset! Will release in the next few days!

**Comment by rowanc1** [comment]
Thanks for your help with this @eurunuela!

**Comment by eurunuela** [comment]
Thank you @rowanc1 for the help!

**Comment by eurunuela** [comment]
> Also for testing:
> 
> ```yaml
> site:
>   template: ../path/to/myst-theme/book
> ```
> 
> This is a bit annoying for testing as even when running in `myst start --headless` it still reads the local template options (which doesn't have the `logo_dark`) to know to copy the file over, etc.

Either I'm doing something wrong or the snippet above doesn't work. I get this error:

```
Problem with template link "https://api.mystmd.org/templates//Users/eurunuela/GitHub/myst-theme": 404 Not Found

To list valid templates, try the command "myst templates list"
```

**Comment by eurunuela** [comment]
Hey @rowanc1, I'm still having the issue with the logo:

`'options' extra key ignored: logo_dark (at myst.yml)`

I updated to the latest `mystmd` version, so I don't know if there is something in my setup that is making it fail. Any ideas?

**Comment by rowanc1** [comment]
I think that you are on a previous template. Can you try:

```
myst clean --templates
```

And then re-run `myst start`?

**Comment by eurunuela** [comment]
Agh cleaning did it! Thank you Rowan!

Maybe the cleaning command could happen when updating myst?

**Comment by rowanc1** [comment]
Yeah, we need to document this better and have some indication that the templates are out of date. Right now we are skirting versioning on the templates and starting to be a bit of an issue in some cases.

Adding a quick note on this in [#467](https://github.com/executablebooks/mystmd/pull/467).

**Pull request merged**